lib/async_hooks.js

```diff
@@ -461,6 +461,21 @@ function init(asyncId, type, triggerId, resource) {
   processing_hook = false;
 }
 
+const JSSideProviders = {
+  Timeout: 'timers Timeout',
+  Immediate: 'timers Immediate',
+  TickObject: 'next_tick TickObject'
+};
+
+function _getTypeName(module, type) {
+  return Object.keys(JSSideProviders)
+    .find((k) => JSSideProviders[k] === `${module} ${type}`);
+}
```

Comment:
> The question is will it be robust (no name conflicts)

That depends on what exactly your idea is – just have this for timers and nextTick? That doesn’t seem much more useful than listing these as values. If you’re trying to do this for generic async providers, you’ll run into the problem that many public APIs (e.g. `net.Socket`) can init different async types.

> that way it's also useful to end users.

I wouldn’t underscore-prefix it in that case